packages/siwe/lib/types.ts

```diff
@@ -26,14 +26,17 @@ export interface VerifyOpts {
   /** ethers provider to be used for EIP-1271 validation */
   provider?: providers.Provider;
 
-    /** If the library should reject promises on errors, defaults to false */
-    suppressExceptions?: boolean;
+  /** If the library should reject promises on errors, defaults to false */
+  suppressExceptions?: boolean;
 
-    /** Function to be called when the message cannot be validated. */
-    failureCallback?: (params: VerifyParams, opts: VerifyOpts, message: SiweMessage) => Promise<SiweResponse>;
+  /** Enables a custom verification function that will be ran alongside EIP-1271 check. */
+  verificationFallback?: (params: VerifyParams, opts: VerifyOpts, message: SiweMessage, EIP1271Promise: Promise<SiweResponse>) => Promise<SiweResponse>;
 }
 
-export const VerifyOptsKeys: Array<keyof VerifyOpts> = ["provider", "suppressExceptions", "failureCallback"];
+export const VerifyOptsKeys: Array<keyof VerifyOpts> = [
+  'provider',
+  'suppressExceptions',
+];
```

Comment:
Does `verificationFallback` need to be here?

Reply:
good catch.